<?xml version="1.0" encoding="utf-8" ?>
<launch>
    <arg name="bagPath" default="/seerep/seerep-data/claas/07-20-12-12-03_agri-gaia_ap4-4.bag"/>
    <arg name="classesMappingPath" default="/seerep/seerep-data/rosbag-FR/20230524/label_seerep_fr.json"/>
    <arg name="hdf5FolderPath" default="/seerep/seerep-data/claas"/>
    <arg name="projectFrameId" default="map"/>
    <arg name="projectName" default="testproject"/>
    <arg name="topicImage" default="/sensorbox/mekra100/image_raw"/>
    <arg name="topicCameraIntrinsics" default="/sensorbox/mekra100/camera_info"/>
    <arg name="topicDetection" default="/cama/classifier_gpu_node/cw_detection_array"/>
    <arg name="detectionCategory" default="theCategory"/>
    <arg name="topicTf" default="/tf"/>
    <arg name="topicTfStatic" default="/tf_static"/>
    <arg name="topicGeoAnchor" default="/anchor"/>
    <arg name="distanceCameraGround" default="0.5"/>
    <arg name="maxViewingDistance" default="15.0"/>
    <arg name="storeImages" default="true"/>
    <!-- <arg name="projectUuid" default=""/> -->

    <node name="seerep_ros_communication_rosbagDumper" pkg="seerep_ros_communication" type="seerep_ros_communication_rosbagDumper" output="screen">
        <rosparam param="bagPath" subst_value="True">$(arg bagPath)</rosparam>
        <rosparam param="classesMappingPath" subst_value="True">$(arg classesMappingPath)</rosparam>
        <rosparam param="hdf5FolderPath" subst_value="True">$(arg hdf5FolderPath)</rosparam>
        <rosparam param="projectFrameId" subst_value="True">$(arg projectFrameId)</rosparam>
        <rosparam param="projectName" subst_value="True">$(arg projectName)</rosparam>
        <rosparam param="topicImage" subst_value="True">$(arg topicImage)</rosparam>
        <rosparam param="topicCameraIntrinsics" subst_value="True">$(arg topicCameraIntrinsics)</rosparam>
        <rosparam param="topicDetection" subst_value="True">$(arg topicDetection)</rosparam>
        <rosparam param="detectionCategory" subst_value="True">$(arg detectionCategory)</rosparam>
        <rosparam param="topicTf" subst_value="True">$(arg topicTf)</rosparam>
        <rosparam param="topicTfStatic" subst_value="True">$(arg topicTfStatic)</rosparam>
        <rosparam param="topicGeoAnchor" subst_value="True">$(arg topicGeoAnchor)</rosparam>
        <rosparam param="distanceCameraGround" subst_value="True">$(arg distanceCameraGround)</rosparam>
        <rosparam param="maxViewingDistance" subst_value="True">$(arg maxViewingDistance)</rosparam>
        <rosparam param="storeImages" subst_value="True">$(arg storeImages)</rosparam>
        <!-- <rosparam param="projectUuid" subst_value="True">$(arg projectUuid)</rosparam> -->
    </node>
</launch>
